src/components/data-model-table/association-selection.tsx

```diff
@@ -49,38 +51,53 @@ export const AssociationSelection = <
   } = props;
 
   const dispatch = useDispatch();
-  const storageKey =
-    `${(parentRecord as object).constructor.name}_${associatedRecordClass.name}`.toLowerCase();
-
-  const associatedRecordClassInstance = plainToInstance(
-    associatedRecordClass,
-    {},
+  const storageKey = useMemo(
+    () =>
+      `${(parentRecord as object).constructor.name}_${associatedRecordClass.name}`.toLowerCase(),
+    [parentRecord, associatedRecordClass],
   );
 
-  const associatedRecordResourceType = Reflect.getMetadata(
-    CLASS_RESOURCE_TYPE,
-    associatedRecordClassInstance as object,
+  const associatedRecordClassInstance = useMemo(
+    () => plainToInstance(associatedRecordClass, {}),
+    [associatedRecordClass],
   );
 
-  const selectedIdentifiers =
-    useSelector(
-      getSelectedKeyValue(storageKey, associatedRecordClassInstance as object),
-    ) || '';
+  const associatedRecordResourceType = useMemo(
+    () =>
+      Reflect.getMetadata(
+        CLASS_RESOURCE_TYPE,
+        associatedRecordClassInstance as object,
+      ),
+    [associatedRecordClassInstance],
+  );
 
-  if (!associatedRecordResourceType) {
-    return (
-      <Alert
-        message="Error: AssociationSelection cannot find ResourceType for associatedRecordClass"
-        type="error"
-      />
-    );
-  }
+  const label = useMemo(
+    () =>
+      Reflect.getMetadata(LABEL_FIELD, associatedRecordClassInstance as object),
+    [associatedRecordClassInstance],
+  );
 
-  const primaryKeyFieldName: string = Reflect.getMetadata(
-    PRIMARY_KEY_FIELD_NAME,
-    associatedRecordClassInstance as object,
+  const primaryKeyFieldName = useMemo(
+    () =>
+      Reflect.getMetadata(
+        PRIMARY_KEY_FIELD_NAME,
+        associatedRecordClassInstance as object,
+      ),
+    [associatedRecordClassInstance],
   );
 
+  const selectedItems = useSelector(getSelectedAssociatedItems(storageKey));
+
+  const selectedIdentifiers = useMemo(() => {
```

Comment:
I like that you moved this .join into here, I am not sure if we will be joining it like this in many other places, so having it like this where the selector returns the list of items and the component using the selector does in this case the join, I think makes a lot of sense 👍